Source/DATASource.h

```diff
@@ -87,4 +87,18 @@
      withIndexPath:(NSIndexPath *)indexPath
       newIndexPath:(NSIndexPath *)newIndexPath;
 
+    - (BOOL)dataSource:(UITableView *)tableView
+ canEditRowAtIndexPath:(NSIndexPath *)indexPath;
+
+- (NSString *)dataSource:(UITableView *)tableView titleForDeleteConfirmationButtonForRowAtIndexPath:(NSIndexPath *)indexPath;
```

Comment:
I'm going to need to drop this one since it's part of the `UITableViewDelegate` not the `UITableViewDataSource` so it could easily live in your main controller.